gameplay.js

```diff
@@ -241,6 +250,23 @@ function positionWithinBoard(position){
     return typeof position === "object";
 }
 
+function nextX(x, y, dx, dy){
```

Comment:
Not a descriptive function name - without really looking at the function I have no idea what it does.